src/slist.c

```diff
@@ -23,40 +23,42 @@ CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 */
 
-#include <stdlib.h>
-#include <Imlib2.h>
 #include "slist.h"
+#include <Imlib2.h>
+#include <stdlib.h>
 
-Scrot_Imlib_List * append_to_scrot_imlib(Scrot_Imlib_List *head, Imlib_Image *data) {
-  
-  Scrot_Imlib_List *tail = walk_to_end_of_scrot_imlib_list(head);
-  Scrot_Imlib_List *appended = (Scrot_Imlib_List *) malloc(sizeof(Scrot_Imlib_List));
-  appended->data = data;
-  appended->next = NULL;
+ScrotImlibList* appendToScrotImlib(ScrotImlibList* head, Imlib_Image* data)
+{
+    ScrotImlibList* tail = walkToEndOfScrotImlibList(head);
+    ScrotImlibList* appended;
+    appended = malloc(sizeof(appended));
```

Comment:
This is the size in bytes of a pointer, the correct thing would be: `sizeof(*appended)`

Reply:
oops. will fix that.

Reply:
fixed